libs/langchain/langchain/vectorstores/kdbai.py

```diff
@@ -0,0 +1,243 @@
+from __future__ import annotations
+
+import logging
+import uuid
+from typing import Any, Callable, Iterable, List, Optional, Tuple, Union
+
+import numpy as np
+import pandas as pd
+
+from langchain.docstore.document import Document
+from langchain.embeddings.base import Embeddings
+from langchain.vectorstores.base import VectorStore
+from langchain.vectorstores.utils import DistanceStrategy, maximal_marginal_relevance
+
+logger = logging.getLogger(__name__)
+
+
+class KDBAI(VectorStore):
+    """"`KDB.AI` vector store [https://kdb.ai](https://kdb.ai)
+    
+    To use, you should have the `kdbai_client` python package installed.
+
+    Args:
+        table: kdbai_client.Table object to use as storage,
+        embedding: Any embedding function implementing
+            `langchain.embeddings.base.Embeddings` interface,
+        distance_strategy: One option from DistanceStrategy.EUCLIDEAN_DISTANCE, 
+            DistanceStrategy.DOT_PRODUCT or DistanceStrategy.COSINE.
+
+    See the example [notebook](https://github.com/KxSystems/langchain/blob/KDB.AI/docs/docs/integrations/vectorstores/kdbai.ipynb).
+    """
+
+    def __init__(
+        self,
+        table: Any,
+        embedding: Union[Embeddings, Callable],
```

Comment:
preference to just take Embeddings (any vecstores that take both do so for vestigial + backwards compatibility reasons)

Reply:
Thanks @baskaryan for the speedy review, I will proceed with the changes in the next couple hours.

Reply:
Should be good following new commit:  https://github.com/langchain-ai/langchain/pull/12797/commits/ed5834c9520482f3a88d7de8983f74847906e84d